core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -131,4 +131,16 @@ public void testDataFilePositions() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testDeleteFilePositions() throws IOException {
```

Comment:
The unit test looks good to me.

Reply:
I'd like to provide an Assume here so that we don't have to test this case for format version 1 .  (Actually, the current patch will test the same case twice for both format version 1 and 2,  we don't need to do that).

```java
Assume.assumeTrue("Delete files only work for format version 2", formatVersion == 2);
```